src/Interpreters/MarkTableIdentifiersVisitor.cpp

```diff
@@ -23,37 +43,22 @@ bool MarkTableIdentifiersMatcher::needChildVisit(ASTPtr & node, const ASTPtr & c
 void MarkTableIdentifiersMatcher::visit(ASTPtr & ast, Data & data)
 {
     if (auto * node_func = ast->as<ASTFunction>())
-        visit(*node_func, ast, data);
+        visit(*node_func, data);
 }
 
-void MarkTableIdentifiersMatcher::visit(const ASTFunction & func, ASTPtr & ptr, Data & data)
+void MarkTableIdentifiersMatcher::visit(const ASTFunction & func, const Data & data)
 {
     /// `IN t` can be specified, where t is a table, which is equivalent to `IN (SELECT * FROM t)`.
     if (checkFunctionIsInOrGlobalInOperator(func))
     {
-        auto ast = func.arguments->children.at(1);
-        auto opt_name = tryGetIdentifierName(ast);
-        if (opt_name && !data.aliases.count(*opt_name) && ast->as<ASTIdentifier>())
-        {
-            ptr->as<ASTFunction>()->arguments->children[1] = ast->as<ASTIdentifier>()->createTable();
-            assert(ptr->as<ASTFunction>()->arguments->children[1]);
-        }
+        replaceArgumentWithTableIdentifierIfNotAlias(func, 1, data.aliases);
     }
 
     // First argument of joinGet can be a table name, perhaps with a database.
     // First argument of dictGet can be a dictionary name, perhaps with a database.
     else if (functionIsJoinGet(func.name) || functionIsDictGet(func.name))
     {
-        if (!func.arguments || func.arguments->children.empty())
-            return;
-
-        auto ast = func.arguments->children.at(0);
-        auto opt_name = tryGetIdentifierName(ast);
-        if (opt_name && !data.aliases.count(*opt_name) && ast->as<ASTIdentifier>())
-        {
-            ptr->as<ASTFunction>()->arguments->children[0] = ast->as<ASTIdentifier>()->createTable();
```

Comment:
The crash was here because `createTable()` returned nullptr and `assert()` didn't alert on that in release.